oci/table_oci_queue_queue.go

```diff
@@ -0,0 +1,336 @@
+package oci
+
+import (
+	"context"
+	"strings"
+
+	oci_common "github.com/oracle/oci-go-sdk/v65/common"
+	"github.com/oracle/oci-go-sdk/v65/queue"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/turbot/go-kit/types"
+
+	"github.com/turbot/steampipe-plugin-sdk/v4/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v4/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/v4/plugin/transform"
+)
+
+//// TABLE DEFINITION
+
+func tableQueueQueue(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "oci_queue_queue",
+		Description: "OCI Queue Queue",
+		Get: &plugin.GetConfig{
+			KeyColumns: plugin.SingleColumn("id"),
+			Hydrate:    getQueue,
+		},
+		List: &plugin.ListConfig{
+			Hydrate: listQueues,
+			KeyColumns: []*plugin.KeyColumn{
+				{
+					Name:    "compartment_id",
+					Require: plugin.Optional,
+				},
+				{
+					Name:    "display_name",
+					Require: plugin.Optional,
+				},
+				{
+					Name:    "id",
+					Require: plugin.Optional,
+				},
+				{
+					Name:    "lifecycle_state",
+					Require: plugin.Optional,
+				},
+			},
+		},
+		GetMatrixItemFunc: BuildCompartementRegionList,
+		Columns: []*plugin.Column{
+			{
+				Name:        "display_name",
+				Description: "The name of the queue.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("DisplayName"),
+			},
+			{
+				Name:        "id",
+				Description: "The OCID of the queue.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromCamel(),
+			},
+			{
+				Name:        "messages_endpoint",
+				Description: "The endpoint to use to get or put messages in the queue.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromCamel(),
+			},
+			{
+				Name:        "lifecycle_details",
+				Description: "A message describing the current state in more detail. For example, can be used to provide actionable information for a resource in Failed state.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromCamel(),
+			},
+			{
+				Name:        "lifecycle_state",
+				Description: "The current state of the Queue.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromCamel(),
+			},
+			{
+				Name:        "time_created",
+				Description: "The date and time the queue was created.",
+				Type:        proto.ColumnType_TIMESTAMP,
+				Transform:   transform.FromField("TimeCreated.Time"),
+			},
+			{
+				Name:        "time_updated",
+				Description: "The date and time the queue was last modified.",
+				Type:        proto.ColumnType_TIMESTAMP,
+				Transform:   transform.FromField("TimeUpdated.Time"),
+			},
+			{
+				Name:        "retention_in_seconds",
+				Description: "The retention period of the messages in the queue, in seconds.",
+				Hydrate:     getQueue,
+				Type:        proto.ColumnType_INT,
+				Transform:   transform.FromCamel(),
```

Comment:
@lucasjellema instead of defining `transform.FromCamel()` for every column, it is safe to define a default transform for this table.

Please check https://github.com/turbot/steampipe-plugin-oci/blob/main/oci/table_oci_kms_key.go#L23 for more information.

Reply:
Thanks for the tip . I will update the code to that effect